pallets/file-system/src/lib.rs

```diff
@@ -313,12 +394,27 @@ pub mod pallet {
             let start_block = NextStartingBlockToCleanUp::<T>::get();
             let mut block_to_clean = start_block;
 
-            // Total weight used to avoid exceeding the remaining weight
+            // Total weight used to avoid exceeding the remaining weight.
+            // We count one write for the `NextBlockToCleanup` storage item updated at the end.
             let mut total_used_weight = Weight::zero();
 
             let required_weight_for_iteration =
                 db_weight.reads_writes(1, T::MaxExpiredStorageRequests::get().into());
 
+            // Check if we can process a single iteration and update the `NextStartingBlockToCleanUp` storage item
+            if !remaining_weight.all_gte(
```

Comment:
I think this generates more confusion in the flow of the function, than the benefits it brings. Think of it, the check of whether `remaining_weight <= total_used_weight + required_weight_for_iteration` is already performed in the beginning of the `while` clause. This "pre-check" is not really needed. Because if that is the case, what's inside the `while` will never be executed, `block_to_clean` will never be modified, so the function will not do anythng besides checking the `while` clause and the `if` afterwards, which is not a big deal at all.

Reply:
Yes I agree, but the problem with this logic is that we check everything but the final write at the end, and it is possible that just a single write can have the `total_used_weight` exceed the `remaining_weight`. Hence we have to check here at the beginning of everything we can execute the entire `on_idle` function with all reads and writes that are possibly going to happen. It is standard practice to do this for hook executions.

Reply:
After brief discussion, we can include the write in the `required_weight_for_iteration` since it will not mutate the `total_used_weight` during iteration but will still check for the final write.